ping.go

```diff
@@ -587,6 +590,17 @@ func (p *Pinger) sendICMP(conn *icmp.PacketConn) error {
 		}
 		p.PacketsSent++
 		p.sequence++
+
+		handler := p.OnSend
+		if handler != nil {
+			outPkt := &Packet{
+				Nbytes: len(msgBytes),
+				IPAddr: p.ipaddr,
+				Addr:   p.addr,
+				Seq:    p.sequence - 1,
+			}
+			handler(outPkt)
```

Comment:
I did wonder whether this should be asynchronous `go handler(outPkt)` but it matches the other `OnRecv` event handlers we have in the code. We can look at this in a future PR perhaps.

Reply:
Yes, I modeled the behaviour after the OnRecv handler.